packages/comlink/src/controller.ts

```diff
@@ -260,8 +260,7 @@ export const createController = (input: {targetOrigin: string}): Controller => {
 
     const stop = () => {
       channels.forEach((channel) => {
-        channel.disconnect()
-        channel.stop()
+        cleanupChannel(channel as unknown as Channel<Message, Message>)
```

Comment:
Will the following also work?
```suggestion
        cleanupChannel(channel)
```

Reply:
Unfortunately not as it stands, I guess this has to do with type variance? `channel` here is of type `Channel<R, S>`, and that isn't considered a subtype of `Channel<Message, Message>` even if `R` and `S` extend `Message`...or something like that.

I need to look at improving the typing in places, if you have any suggestions they'd be very welcome! 😊